app/tasks/raster_tile_cache_assets/raster_tile_cache_assets.py

```diff
@@ -126,6 +126,13 @@ async def raster_tile_cache_asset(
     assert source_asset_co.symbology is not None
     symbology_function = symbology_constructor[source_asset_co.symbology.type]
 
+    # We want to make sure that the final RGB asset is called after the implementation of the tile cache
+    if symbology_function == no_symbology:
+        source_asset_co.pixel_meaning = implementation
```

Comment:
Following this logic through out the background task now. 
The final asset used to generate the tile cache will have same name as tile cache itself.
TODO: I might need to add some additional logic to catch cases where the tile cache implementation is called after an intermediate step, which would cause data overwrite and possible data corruption!